ci/check_no-duplicate-symbols.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/python3
```

Comment:
This doesn't seem to be executed anywhere on CI?

Reply:
I thought it would be better to place the script somewhere other than the top-level directory, but then after I moved it into `ci/` I noticed that `run.sh` already has a duplicate symbols check. My script prints out some more details though, to help debug the situation.

Reply:
Oh sure yeah, but I didn't actually see this run anywhere on CI, and there's also not any documentation as to what this script does, so I wasn't sure why it was added.

Reply:
Yeah, `ci/` is probably not the most suitable place for it. Shall I move it back up to the top-level directory? Or something like `dev/` or `scripts/`

Reply:
Er sorry my point isn't where this should go, it's that I still don't know what this does, how it's supposed to be run, or what it runs on. If it's not run on CI then it's probably best to leave it out for now.

Reply:
OK, I've removed it from this PR.